### PR TITLE
feat: add JobCategorySelect component

### PR DESCRIPTION
**Pull request by aaronvanston**
## Overview
Pulls in the JobCategorySelect. The JobCategory select component provides a full list of parent and child job categories for SEEK. Returns a JobCategory (parent or child) with object identifier on category select.

This component can be used to fetch a job category or used inline within a form that generates the input structure for other queries and mutations. For example, you can use this component to resolve a job category OID for PositionPosting.

## Includes:
- Storybook example, only resolves a mock query with 2 job categories and 2 child job categories each
- Readme with usage instructions.

### CODE REVIEW

File: fe/lib/components/JobCategorySelect/queries.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    }
+  }
+  ${NESTED_JOB_CATEGORY_ATTRIBUTES}
+`;
+
+export const JOB_CATEGORY_SUGGESTION = gql`
+  query(
+    $positionProfile: JobCategorySuggestionPositionProfileInput!
+    $schemeId: String!
+    $hirerId: String
```
Comment (etaoins) [inline fe/lib/components/JobCategorySelect/queries.ts:50]: I was waiting for this PR so I could make someone else do my work!

Can we move this `hirerId` in to `positionProfile.positionOrganizations` as per SEEK-Jobs/indie-graphql-server#596

Comment (aaronvanston) [reply fe/lib/components/JobCategorySelect/queries.ts:50]: Yeah, sure thing! Once this is merged back into our internal repo's I'll remove their usage of the deprecated structure.

Comment (aaronvanston) [reply fe/lib/components/JobCategorySelect/queries.ts]: I've just removed it from the variables for this query. I don't actually use this query yet just copied the whole lot for now. My next PR will add JobCategoriesSuggest which will, I'll use the updated `JobCategorySuggestionPositionProfileInput` shape with hirerid there.

File: fe/lib/components/JobCategorySelect/README.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# JobCategorySelect
+
+A JobCategory select component that provides a full list of parent and child job categories for SEEK. Returns a JobCategory (parent or child) with object identifier on category select.
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/README.md:3]:
```suggestion
A JobCategory select component that provides a full list of parent and child job categories for SEEK.
Returns a JobCategory (parent or child) with object identifier on category selection.
```

File: fe/lib/components/JobCategorySelect/README.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## Installation
+
+```shell
+$ yarn add wingman-fe
+```
+
+## Job Category Select Widget
+
+The JobCategorySelect widget abstracts the `jobCategories` query on SEEK API and provides a select input that returns a SEEK JobCategory. Read more about underlying [`jobCategories` query](https://developer.seek.com/schema/#operation-jobCategories).
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/README.md:13]:
```suggestion
The JobCategorySelect widget abstracts the `jobCategories` query on SEEK API and provides a select input that returns a SEEK JobCategory.
Read more about underlying [`jobCategories` query](https://developer.seek.com/schema/#operation-jobCategories).
```

File: fe/lib/components/JobCategorySelect/README.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+### Properties
+
+Required
+
+- `schemeId`: The scheme for the job category dataset to query.
+
+Optional:
+
+- `client`: An `ApolloClient` instance. By default JobCategorySelect uses the client passed down via context, but a different client can be passed in.
+- `onSelect`: Callback function that is supplied a [SEEK JobCategory](https://developer.seek.com/schema/#definition-JobCategory) on job category select.
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/README.md:24]:
```suggestion
- `onSelect`: Callback function that is supplied a [SEEK JobCategory](https://developer.seek.com/schema/#definition-JobCategory) on job category selection.
```

File: fe/lib/components/JobCategorySelect/JobCategorySelect.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      error: categoriesError,
+    } = useQuery(JOB_CATEGORIES, {
+      ...(client && { client }),
+      fetchPolicy: 'no-cache',
+      variables: {
+        schemeId,
+      },
+    });
+
+    const [selectedJobCategoryId, setselectedJobCategoryId] = useState('');
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/JobCategorySelect.tsx:39]: 😛

```diff
- setselectedJobCategoryId
+ setSelectedJobCategoryId
```

File: fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        selectedParentCategoryId,
+      );
+
+      if (parentCategory?.children) {
+        setChildCategories(parentCategory.children);
+        setSelectedChildCategoryId('');
+        onSelect(parentCategory);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx:34]: Just for my benefit, which deps are problematically changing between renders here?

File: fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+      );
+      if (childCategory) {
+        onSelect(childCategory);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedChildCategoryId, childCategories]);
+
+  return (
+    <Stack space="medium">
```
Comment (72636c) [inline fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx:51]: Redundant Stack?

File: fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx (hunk 6 above)
Comment (aaronvanston) [reply fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx:34]: It had re-rendering issuers with `setChildCategories`, `setSelectedChildCategoryId` and `onSelect`. It was causing extraneous renders. We could look into using useCallback in the future to see if they can be mitigated.

**Pull request merged**